envi/archs/i386/disasm.py

```diff
@@ -666,16 +668,16 @@ def __init__(self, mode=MODE_32):
         self._dis_amethods[opcode86.ADDRMETH_V>>16] = self.ameth_v
         self._dis_amethods[opcode86.ADDRMETH_X>>16] = self.ameth_x
         self._dis_amethods[opcode86.ADDRMETH_Y>>16] = self.ameth_y
-        self._dis_amethods[opcode86.ADDRMETH_Z>>16] = self.ameth_z
 
         # Offsets used to add in addressing method parsers
-        self.ROFFSETMMX   = getRegOffset(i386regs, "mm0")
+        # MMX is just a meta reg of st
         self.ROFFSETSIMD  = getRegOffset(i386regs, "xmm0")
         self.ROFFSETDEBUG = getRegOffset(i386regs, "debug0")
         self.ROFFSETCTRL  = getRegOffset(i386regs, "ctrl0")
         self.ROFFSETTEST  = getRegOffset(i386regs, "test0")
         self.ROFFSETSEG   = getRegOffset(i386regs, "es")
         self.ROFFSETFPU   = getRegOffset(i386regs, "st0")
+        self.ROFFSETMMX   = self._dis_regctx.getRegisterIndex("mm0")
```

Comment:
Since the MMX registers are described as meta registers of the floating point registers, I made that change, and with that, getRegOffset complains that it doesn't know about the new meta register mm0.

Figured it'd be a breakage of how we usually do things if we suddenly changed things to have getRegOffset return metaregs

Reply:
if it's not a performance hit for most things, it might be worth considering.
otherwise, i will typically just point the self.ROFFSETMMX = self.<whatever_the_real_register_base_is>... or just use the real register base in place of ROFFSETMMX.  or does that not work in this case?

Reply:
Just so I'm clear, you're proposing: `self.ROFFSETMMX = getRegOffset(i386regs, 'st0')` or something different? Because the former would fail a bunch of tests that we've got for i386 and amd64.

Reply:
no, i get it now.  it's ugly.  slap a comment in front of it to let people know why.

Reply:
:+1: